cli/src/commands/config/set.rs

```diff
@@ -46,7 +54,63 @@ pub fn cmd_config_set(
             path = config_path.display()
         )));
     }
+
     // TODO(#531): Infer types based on schema (w/ --type arg to override).
     let value = parse_toml_value_or_bare_string(&args.value);
+
+    // If the user is trying to change the author config, we should warn them that
+    // it won't affect the working copy author
+    if args.name == ConfigNamePathBuf::from_iter(vec!["user", "name"]) {
+        check_wc_author(command, ui, &value, AuthorChange::Name)?;
+    } else if args.name == ConfigNamePathBuf::from_iter(vec!["user", "email"]) {
+        check_wc_author(command, ui, &value, AuthorChange::Email)?;
+    };
+
     write_config_value_to_file(&args.name, value, &config_path)
 }
+
+/// Returns the commit of the working copy if it exists.
+fn maybe_wc_commit(helper: &WorkspaceCommandHelper) -> Option<Commit> {
+    let repo = helper.repo();
+    let maybe_wc_commit = helper
+        .get_wc_commit_id()
+        .map(|id| repo.store().get_commit(id))
+        .transpose()
+        .unwrap();
+    maybe_wc_commit
+}
+
+/// Check if the working copy author name matches the user's config value
+/// If it doesn't, print a warning message
+fn check_wc_author(
+    command: &CommandHelper,
+    ui: &mut Ui,
+    new_value: &toml_edit::Value,
+    author_change: AuthorChange,
+) -> Result<(), CommandError> {
```

Comment:
nit: let's restrict the error type to `io::Result<()>` to ensure that `workspace_helper()` error is suppressed.